test/pkg/integration/workspace.go

```diff
@@ -683,21 +690,33 @@ func WaitForWorkspaceStart(t *testing.T, ctx context.Context, instanceID string,
 		wsman, err := api.WorkspaceManager()
 		if err != nil {
 			api.ClearWorkspaceManagerClientCache()
-			return nil, true, nil
+			return nil, false, err
 		}
 		desc, err := wsman.DescribeWorkspace(ctx, &wsmanapi.DescribeWorkspaceRequest{
 			Id: instanceID,
 		})
 		if err != nil {
 			scode := status.Code(err)
 			if scode == codes.NotFound || strings.Contains(err.Error(), "not found") {
+				if cfg.WaitForStopped {
+					t.Logf("describe: workspace couldn't be found, but we're expecting it to stop, so wait for subscribe to give us the last status")
+					return nil, false, nil
+				}
 				if !cfg.CanFail {
-					return nil, false, xerrors.New("the workspace couldn't find")
+					return nil, true, xerrors.New("the workspace couldn't be found")
```

Comment:
Why true, instead of false? I ask because Not Found could indicate the pod hasn't been scheduled yet, I thought.

Reply:
`WaitForWorkspaceStart` only gets called after creating the workspace, so if it can't be found it should be safe to assume it has disappeared in the meantime. Main reason for the change was that in some tests the workspace actually did disappear and having it at `false` meant the test would continue waiting for the workspace until the context times out, setting it to `true` makes it fail faster